dumper/CMakeLists.txt

```diff
@@ -1,12 +1,13 @@
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${dist_dir}/dumper)
 
-add_executable(cose_dumper dumper.c)
+add_executable(cose_dumper dumper.cpp)
 target_link_libraries(cose_dumper PRIVATE cn-cbor::cn-cbor cose-c::cose-c)
 if(MSVC)
   target_link_libraries(cose_dumper PRIVATE ws2_32)
 endif()
 
 target_include_directories(cose_dumper PRIVATE ../src)
+target_compile_options(cose_dumper PRIVATE "-fpermissive")
```

Comment:
as we are using cpp, we can overload functions -> we should get rid if macros that insert `, context` to the argument list IMHO

Reply:
It is not going to be possible to remove all of these macros as the CBOR library is still in C.  I am also not too sure how much one can rely on linkers to not require functions exist, I know that one can compile each function into a separate block for MSVC but I don't know for sure that this exists in gcc and clang as they are not the compilers that I use much